### ISSUE TITLE
Multiple Stylesheets break anchor links

### ISSUE THREAD

**Issue by dvschultz** [dvschultz/99problems#31]
from @makeBetterEBooks here: https://github.com/dvschultz/99problems/issues/22#issuecomment-47493764

Some more fun and games, in ADE 1.7 if multiple CSS sheets are referenced by the same HTML page, internal hyperlinks will go to the top of that HTML page, NOT the specific anchor.

Never did extensive testing, so unsure if it was the presence of multiple CSS sheets, or the fact that the second one was for mobi.

**Comment by teytag** [comment]
Hi,

My solutions;

![screen shot 2014-07-10 at 12 58 26 pm](https://cloud.githubusercontent.com/assets/6927098/3536993/5e3284e8-0819-11e4-901b-3165ed3525b8.png)
![screen shot 2014-07-10 at 12 54 00 pm](https://cloud.githubusercontent.com/assets/6927098/3536995/69a71550-0819-11e4-8e59-1465f5f6002b.png)
![screen shot 2014-07-10 at 12 56 52 pm](https://cloud.githubusercontent.com/assets/6927098/3536996/69f07ad8-0819-11e4-8265-f7b2856e96a8.png)

**Comment by dvschultz** [comment]
@teytag I don’t think this answers the problem: when multiple stylesheets exist, anchor links are broken. This shows you _can_ load multiple stylesheets, but not that anchor links work.

**Comment by teytag** [comment]
@dvschultz This error may be momentary. I think this error is not multiply sourced stylesheet. Could you show your codes?

**Comment by JayPanoz** [comment]
It's actually worse… (and it still lives in BlueFire for instance)

Say you've got an HTML file A with a link to another HTML File B: 
1. A.html got one stylesheet, ids are used for callback
2. B.html got two stylesheets, href points to A.html#id
3. when you click B.html to go back to A.html#id, you're back at the top of A.html, not at the id.

So it doesn't only impact one HTML file but 2 when 1 of them got multiple stylesheets.

**Comment by tooolbox** [comment]
@JayPanoz Curious how this relates to #22 and the workaround that it has--putting Media Queries in a separate CSS file.  So if you do that, your hyperlinks are toast?

Just want to make sure I understand.

**Comment by JayPanoz** [comment]
I've actually experienced that this morning for an ePub 3 file as I usually put everything “progressive enhancement” is a separate CSS file. So: 
- there was a link in the xhtml file with only 1 stylesheet to the file with 2 stylesheets;
- there was a link back in this file with 2 stylesheets, it was heading to an id in the first file;
- tested it back to ADE 1.8 and everything was OK;
- tested it in some apps using RMSDK and the bug applied (e.g. Bluefire Reader iOS, etc.) → instead of heading to the id, it headed to the top of the xhtml file where the id was.

Now, the file with the id got one single stylesheet… only the other one got 2. So it also impacts a file with one stylesheet if the link is in another xhtml file which got 2. 

![bug](https://cloud.githubusercontent.com/assets/12599652/15789640/9b116be0-29cd-11e6-9b2f-519b513ab427.jpg)
